### PR TITLE
docs: break down Thinking in Loopback

### PR DESCRIPTION
**Pull request by shimks**
- Breaking down Thinking in Loopback into Best Practices with Loopback 4.
  - Remove from Getting Started and have it show at the top level.
  - Create Best Practices summarizing each sections of Thinking in Loopback
  - Broken down into
    - Defining and Validating the API
    - Testing the API
    - Implementing Features
    - Preparing for API Consumption
- connect to https://github.com/strongloop/loopback-next/issues/638

### CODE REVIEW

File: pages/en/lb4/Preparing-API-for-consumption.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+lang: en
+title: 'Preparing API for Consumption'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Preparing-API-for-consumption.html
+summary:
+---
+
+## Preparing your API for consumption
```
Comment (jannyHou) [inline pages/en/lb4/Preparing-API-for-consumption.md]: nitpick: maybe add some note to remind people to follow previous steps.

e.g. In this LB3 tutorial: http://loopback.io/doc/en/lb3/Getting-started-with-LoopBack.html
There is a note on page "Use API Explorer": http://loopback.io/doc/en/lb3/Use-API-Explorer.html

>  Note: If you followed Create a simple API, keep that app running and skip down to Run API Explorer.
If you’re just jumping in, follow the steps below to catch up…

**Review by jannyHou — APPROVED** [review]
LGTM just a nitpick.

File: pages/en/lb4/Best-practices-with-Loopback-4.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+permalink: /doc/en/lb4/Best-practices-with-Loopback-4.html
+summary:
+---
+
+LoopBack 4 is more than just a framework: It’s an ecosystem that encourages developers to follow best practices through predefined standards. This section will walk through some important guidelines by building an example API for a catalog of products.
+
+
+The practice follows an "API first" and test-driven development approach:
+
+1. [**Defining and Validating the API**](./Defining-and-validating-the-API.html): The API is defined first before any internal logic is added to your application.
```
Comment (b-admike) [inline pages/en/lb4/Best-practices-with-Loopback-4.md]: I suggest re-phrasing as follows:
```
This section guides you to construct your API before adding any internal logic to your application.
```
or just
```
Construct your API before adding any internal logic to your application.
```

File: pages/en/lb4/Best-practices-with-Loopback-4.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+2. [**Testing the API**](./Testing-the-API.html): The smoke test for your API and its spec is written.
```
Comment (b-admike) [inline pages/en/lb4/Best-practices-with-Loopback-4.md]: ditto: I suggest rewording it to be an active sentence

File: pages/en/lb4/Best-practices-with-Loopback-4.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+3. [**Implementing Features**](./Implementing-features.html): The tests for each feature of your application are written, and then logic to make these tests pass is written. In the example, the tests for the controller, model, repository, data source, and sequence are written and then implemented.
```
Comment (b-admike) [inline pages/en/lb4/Best-practices-with-Loopback-4.md]: ditto

**Review by b-admike — COMMENTED** [review]
@shimks Looks great! The organization makes sense to me. I've left some comments re wording.

**Comment by crandmck** [comment]
Please edit the titles (both in the sidebar nav and in each individual page) to be sentence case per the style guide: http://loopback.io/doc/en/contrib/Doc-style-guide.html#headings

File: _data/sidebars/lb4_sidebar.yml [hunk 5]
```diff
@@ -85,6 +81,27 @@ children:
...
+
+  - title: 'Testing the API'
+    url: Testing-the-API.html
+    output: 'web, pdf'
+
+  - title: 'Implementing Features'
+    url: Implementing-features.html
+    output: 'web, pdf'
+
+  - title: 'Preparing API for Consumption'
```
Comment (crandmck) [inline _data/sidebars/lb4_sidebar.yml]: "Preparing the API for consumption"

**Review by crandmck — APPROVED** [review]

**Review by b-admike — APPROVED** [review]

File: pages/en/lb4/Testing-the-API.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+complete: Tests took 21ms
+```
+
+It's a powerful proposition to use the API specification not only for API declaration but for test case declaration.  The discussion so far paves the road to "automated controller wireframe-code generation and test-driven development" based on the OpenAPI standard.
+
+At this point, you are ready to make these tests pass by coding up your business logic.
+
+Please refer to [Perform an auto-generated smoke test of your REST API](Testing-your-application.html#perform-an-auto-generated-smoke-test-of-your-rest-api) from [Testing your application](Testing-your-application.html) for more details.
+
+## Define your testing strategy
```
Comment (bajtos) [inline pages/en/lb4/Testing-the-API.md]: Can we move this section to a standalone page please?

**Review by bajtos — COMMENTED** [review]
It was sort of an intention to keep Thinking in LoopBack as a single page, to make it easy for people to read it a story/narrative.

Having said that, I agree the current page has grown way too much and should be splitted.

I am proposing to add links to make it easy for people to read these pages in the right order:

 - At the top of the page, add a link pointing to the previous logical story
 - At the bottom of the page, add a link pointing to the next page to read

**Comment by crandmck** [comment]
>  am proposing to add links to make it easy for people to read these pages in the right order:
> - At the top of the page, add a link pointing to the previous logical story
> - At the bottom of the page, add a link pointing to the next page to read

We have an include template for "next page" that's used in Getting started, (e.g. see bottom of page http://loopback.io/doc/en/lb3/Create-a-simple-API.html).  It would be trivial to create a "previous" template that we could use similarly.  Suggest we use these...

**Review by bajtos — APPROVED** [review]
LGTM at high level, I am afraid I don't have bandwidth to review in detail.
